pandasai/smart_datalake/__init__.py

```diff
@@ -524,7 +524,20 @@ def _add_result_to_memory(self, result: dict):
         Args:
             result (dict): The result to add to the memory
         """
-        if result is None:
+        if not isinstance(result, dict):
+            self.logger.log(
+                f"Inappropriate type of 'result' produced by generated code. "
+                f"Expected 'dict', got '{result.__class__.__name__}'",
+                level=logging.WARNING,
+            )
+            return
+
+        if "type" not in result or "value" not in result:
+            self.logger.log(
+                f"Both 'type' and 'value' items should be present in 'result' "
+                f"produced by generated code. Instead it contains the next "
+                f"content:\n{result}"
+            )
```

Comment:
Great catch. I'm not entirely sure this is the place to do this check.

What if before we run `self._add_result_to_memory(result)` on line 508, we execute another method `self._validate_result` and in there we:
- check if both type and value are present
- check if the type is one of the ones accepted

If any of the conditions is not true, we raise an error.

What do you think? @nautics889

Reply:
Thanks for the review, @gventuri.

___

You're right, there is [another validation](https://github.com/gventuri/pandas-ai/blob/3885c085ed0dae74878479e04007824f2e30679f/pandasai/smart_datalake/__init__.py#L469) a little bit before, therefore on [the 508'th line](https://github.com/gventuri/pandas-ai/blob/3885c085ed0dae74878479e04007824f2e30679f/pandasai/smart_datalake/__init__.py#L508) we likely can operate `validation_ok`. So, as turns out, the current PR brings an excessive validation for containing `"type"` and `"value"`. I've already noticed that before actually, but as seemed to me, it requires a bit of refactoring.

Working on it...

Reply:
Yes, great catch! Fortunately we are now working on simplifying the process a little bit. Too many things happen now and we need to improve the way we separate responsibilities.